Framework/MDAlgorithms/src/IntegratePeaksMDHKL.cpp

```diff
@@ -0,0 +1,319 @@
+                       "[H,0,0]," + boost::lexical_cast<std::string>(h - box) +
+                           "," + boost::lexical_cast<std::string>(h + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("AlignedDim1",
+                       "[0,K,0]," + boost::lexical_cast<std::string>(k - box) +
+                           "," + boost::lexical_cast<std::string>(k + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("AlignedDim2",
+                       "[0,0,L]," + boost::lexical_cast<std::string>(l - box) +
+                           "," + boost::lexical_cast<std::string>(l + box) +
+                           "," + boost::lexical_cast<std::string>(gridPts));
+  normAlg->setProperty("FluxWorkspace", flux);
+  normAlg->setProperty("SolidAngleWorkspace", sa);
+  normAlg->setProperty("OutputWorkspace", "mdout");
+  normAlg->setProperty("OutputNormalizationWorkspace", "mdnorm");
+  normAlg->executeAsChildAlg();
+  Workspace_sptr mdout = normAlg->getProperty("OutputWorkspace");
+  Workspace_sptr mdnorm = normAlg->getProperty("OutputNormalizationWorkspace");
+
+  IAlgorithm_sptr alg = createChildAlgorithm("DivideMD");
+  alg->setProperty("LHSWorkspace", mdout);
+  alg->setProperty("RHSWorkspace", mdnorm);
+  alg->setPropertyValue("OutputWorkspace", "out");
+  alg->execute();
+  IMDWorkspace_sptr out = alg->getProperty("OutputWorkspace");
+  return boost::dynamic_pointer_cast<MDHistoWorkspace>(out);
+}
+
+void IntegratePeaksMDHKL::integratePeak(const int neighborPts,
+                                        MDHistoWorkspace_sptr out,
+                                        double &intensity,
+                                        double &errorSquared) {
+  std::vector<int> gridPts;
+  const size_t dimensionality = out->getNumDims();
+  for (size_t i = 0; i < dimensionality; ++i) {
+    gridPts.push_back(static_cast<int>(out->getDimension(i)->getNBins()));
+  }
+
+  double *F = out->getSignalArray();
+  double Fmax = 0;
+  double Fmin = 1e300;
+  double sum = 0.0;
+  for (int i = 0; i < gridPts[0] * gridPts[1] * gridPts[2]; i++) {
+    if (!boost::math::isnan(F[i]) && !boost::math::isinf(F[i]) && F[i] != 0.0) {
+      sum += F[i];
+      if (F[i] < Fmin)
+        Fmin = F[i];
+      if (F[i] > Fmax)
+        Fmax = F[i];
+    }
+  }
+
+  double *SqError = out->getErrorSquaredArray();
+
+  double minIntensity = Fmin + 0.01 * (Fmax - Fmin);
+  int measuredPoints = 0;
+  int peakPoints = 0;
+  double peakSum = 0.0;
+  double measuredSum = 0.0;
+  double errSqSum = 0.0;
+  double measuredErrSqSum = 0.0;
+  for (int Hindex = 0; Hindex < gridPts[0]; Hindex++) {
+    for (int Kindex = 0; Kindex < gridPts[1]; Kindex++) {
+      for (int Lindex = 0; Lindex < gridPts[2]; Lindex++) {
+        int iHKL = Hindex + gridPts[0] * (Kindex + gridPts[1] * Lindex);
+        if (!boost::math::isnan(F[iHKL]) && !boost::math::isinf(F[iHKL])) {
```

Comment:
Is this check equal to [`!std::isfinite(F[iHKL])`](http://www.cplusplus.com/reference/cmath/isfinite/)?